daemons/controld/controld_schedulerd.c

```diff
@@ -187,25 +187,25 @@ pe_subsystem_new(void)
 
     controld_set_fsa_input_flags(R_PE_REQUIRED);
 
-    if (schedulerd_api != NULL) {
-        pcmk_free_ipc_api(schedulerd_api);
-        schedulerd_api = NULL;
-    }
-
-    rc = pcmk_new_ipc_api(&schedulerd_api, pcmk_ipc_schedulerd);
+    if (schedulerd_api == NULL) {
+        rc = pcmk_new_ipc_api(&schedulerd_api, pcmk_ipc_schedulerd);
 
-    if (rc != pcmk_rc_ok) {
-        crm_err("Error connecting to the scheduler: %s", pcmk_rc_str(rc));
-        return false;
+        if (rc != pcmk_rc_ok) {
+            crm_err("Error connecting to the scheduler: %s", pcmk_rc_str(rc));
+            return false;
+        }
     }
 
     pcmk_register_ipc_callback(schedulerd_api, scheduler_event_callback, NULL);
-    rc = pcmk_connect_ipc(schedulerd_api, pcmk_ipc_dispatch_main);
-    if (rc != pcmk_rc_ok) {
-        crm_err("Error connecting to the scheduler: %s", pcmk_rc_str(rc));
-        pcmk_free_ipc_api(schedulerd_api);
-        schedulerd_api = NULL;
-        return false;
+
+    if (!pcmk_ipc_is_connected(schedulerd_api)) {
+        rc = pcmk_connect_ipc(schedulerd_api, pcmk_ipc_dispatch_main);
+        if (rc != pcmk_rc_ok) {
+            crm_err("Error connecting to the scheduler: %s", pcmk_rc_str(rc));
+            pcmk_free_ipc_api(schedulerd_api);
+            schedulerd_api = NULL;
```

Comment:
Just noticed we don't need to free and null the api object here, just leave it disconnected

Also, pcmk_connect_ipc() will return ok if it's already connected, so we really don't even need to check that at all

Reply:
Fixed.  Nice to remove some unneeded checks and simplify the code.